loader/loader_linux.c

```diff
@@ -0,0 +1,458 @@
+                VkPhysicalDeviceProperties2 dev_props2 = (VkPhysicalDeviceProperties2){
+                    .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2, .pNext = (VkBaseInStructure *)&pci_props};
+
+                if (is_vulkan_1_1 && device_is_1_1_capable) {
+                    icd_term->dispatch.GetPhysicalDeviceProperties2(
+                        sorted_group_term[group].internal_device_info[gpu].physical_device, &dev_props2);
+                } else {
+                    icd_term->dispatch.GetPhysicalDeviceProperties2KHR(
+                        sorted_group_term[group].internal_device_info[gpu].physical_device, &dev_props2);
+                }
+
+                sorted_group_term[group].internal_device_info[gpu].pci_domain = pci_props.pciDomain;
+                sorted_group_term[group].internal_device_info[gpu].pci_bus = pci_props.pciBus;
+                sorted_group_term[group].internal_device_info[gpu].pci_device = pci_props.pciDevice;
+                sorted_group_term[group].internal_device_info[gpu].pci_function = pci_props.pciFunction;
+            }
+            loader_log(inst, VULKAN_LOADER_INFO_BIT | VULKAN_LOADER_DRIVER_BIT, 0, "               [%u] %s", gpu,
+                       sorted_group_term[group].internal_device_info[gpu].device_name);
+        }
+
+        // Select default device if set in the environment variable
+        linux_env_var_default_device(inst, sorted_group_term[group].group_props.physicalDeviceCount,
+                                         sorted_group_term[group].internal_device_info);
+
+        // Sort GPUs in each group
+        qsort(sorted_group_term[group].internal_device_info, sorted_group_term[group].group_props.physicalDeviceCount,
+              sizeof(struct LinuxSortedDeviceInfo), compare_devices);
+    }
+
+    // Sort device groups by PCI info
+    qsort(sorted_group_term, group_count, sizeof(struct loader_physical_device_group_term), compare_device_groups);
+
+    if (loader_get_debug_level() & (VULKAN_LOADER_INFO_BIT | VULKAN_LOADER_DRIVER_BIT)) {
+        loader_log(inst, VULKAN_LOADER_INFO_BIT | VULKAN_LOADER_DRIVER_BIT, 0,
+                   "linux_read_sorted_physical_device_groups:  Sorted order:");
+        for (uint32_t group = 0; group < group_count; ++group) {
+            loader_log(inst, VULKAN_LOADER_INFO_BIT | VULKAN_LOADER_DRIVER_BIT, 0, "           Group %u", group);
+            for (uint32_t gpu = 0; gpu < sorted_group_term[group].group_props.physicalDeviceCount; ++gpu) {
+                loader_log(inst, VULKAN_LOADER_INFO_BIT | VULKAN_LOADER_DRIVER_BIT, 0, "               [%u] %s %s", gpu,
+                           sorted_group_term[group].internal_device_info[gpu].device_name,
+                           (sorted_group_term[group].internal_device_info[gpu].default_device ? "[default]" : ""));
+            }
+        }
+    }
+
+    return res;
+}
+
+#endif  // LOADER_ENABLE_LINUX_SORT
```

Comment:
Should add a blank new line at the end of this file. IIRC google requires it so would be good to do it now instead of finding out it broke them later.

Reply:
Sure, why not